Crash due to negative slice length in Text.fill

It's trying to allocate a slice of length -104 (based on `dlv` debugger). I can't remember exactly, but I think I was doing a `Put` when it crashed.

Full stack trace: [edwood-crash.txt](https://github.com/rjkroege/edwood/files/3264042/edwood-crash.txt)

```
panic: runtime error: makeslice: len out of range

goroutine 99 [running]:
panic(0x60dfe0, 0x6910e0)
        /home/big/go/src/runtime/panic.go:662 +0x2c2 fp=0xc00055fbe0 sp=0xc00055fb50 pc=0x42e832
runtime.panicmakeslicelen(...)
        /home/big/go/src/runtime/slice.go:27
runtime.makeslice(0x5fc300, 0xffffffffffffff98, 0xffffffffffffff98, 0x22)
        /home/big/go/src/runtime/slice.go:44 +0xd3 fp=0xc00055fc10 sp=0xc00055fbe0 pc=0x4447a3
main.(*Text).fill(0xc000230118, 0x7f042cb77048, 0xc000346280)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.0/text.go:542 +0x1ac fp=0xc00055fc88 sp=0xc00055fc10 pc=0x5c204c
main.(*Text).deleted(0xc000230118, 0x0, 0x811)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.0/text.go:613 +0x1f6 fp=0xc00055fcc8 sp=0xc00055fc88 pc=0x5c2426
main.(*File).DeleteAt(0xc000206400, 0x0, 0x811)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.0/file.go:408 +0x167 fp=0xc00055fd58 sp=0xc00055fcc8 pc=0x5ad987
main.(*Text).Delete(0xc000230118, 0x0, 0x811, 0x1)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.0/text.go:578 +0x5c fp=0xc00055fd80 sp=0xc00055fd58 pc=0x5c21fc
main.xfidwrite(0xc000400c60)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.0/xfid.go:564 +0xb08 fp=0xc00055ff88 sp=0xc00055fd80 pc=0x5d0638
main.xfidctl(0xc000400c60, 0x69a220, 0xc000010200)
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.0/xfid.go:52 +0x9c fp=0xc00055ffc8 sp=0xc00055ff88 pc=0x5cdbbc
runtime.goexit()
        /home/big/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc00055ffd0 sp=0xc00055ffc8 pc=0x45c011
created by main.xfidallocthread
        /home/fhs/go/pkg/mod/github.com/rjkroege/edwood@v0.2.0/acme.go:618 +0x1d4
```

--- a/text.go
+++ b/text.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"image"
 	"io"
+	"log"
 	"os"
 	"path/filepath"
 	"sort"
@@ -519,20 +520,25 @@ func (t *Text) inSelection(q0 int) bool {
 }
 
 // Fill inserts additional text from t into the Frame object until the Frame object is full.
-func (t *Text) fill(fr frame.SelectScrollUpdater) {
+func (t *Text) fill(fr frame.SelectScrollUpdater) error {
 	// log.Println("Text.Fill Start", t.what)
 	// defer log.Println("Text.Fill End")
 
 	// Conceivably, LastLineFull should be true or would it only be true if there are no more
 	// characters possible?
 	if fr.IsLastLineFull() || t.nofill {
-		return
+		return nil
 	}
 	if t.file.HasUncommitedChanges() {
 		t.TypeCommit()
 	}
 	for {
 		n := t.file.Size() - (t.org + fr.GetFrameFillStatus().Nchars)
+		if n < 0 {
+			log.Printf("Text.fill: negative slice length %v (file size %v, t.org %v, frame nchars %v)\n",
+				n, t.file.Size(), t.org, fr.GetFrameFillStatus().Nchars)
+			return fmt.Errorf("fill: negative slice length %v", n)
+		}
 		if n == 0 {
 			break
 		}
@@ -563,6 +569,7 @@ func (t *Text) fill(fr frame.SelectScrollUpdater) {
 			break
 		}
 	}
+	return nil
 }
 
 // Delete removes runes [q0, q1). The selection values will be

--- a/text_test.go
+++ b/text_test.go
@@ -8,6 +8,8 @@ import (
 	"reflect"
 	"strings"
 	"testing"
+
+	"github.com/rjkroege/edwood/internal/frame"
 )
 
 func emptyText() *Text {
@@ -234,5 +236,29 @@ func TestGetDirNamesNil(t *testing.T) {
 	_, err := getDirNames(nil)
 	if err == nil {
 		t.Errorf("getDirNames was successful on nil File")
+	}
+}
+
+type textFillMockFrame struct {
+	*MockFrame
+}
+
+func (fr *textFillMockFrame) GetFrameFillStatus() frame.FrameFillStatus {
+	return frame.FrameFillStatus{
+		Nchars:         100,
+		Nlines:         0,
+		Maxlines:       0,
+		MaxPixelHeight: 0,
+	}
+}
+
+func TestTextFill(t *testing.T) {
+	text := &Text{
+		file: &File{},
+	}
+	err := text.fill(&textFillMockFrame{})
+	wantErr := "fill: negative slice length -100"
+	if err == nil || err.Error() != wantErr {
+		t.Errorf("got error %q; want %q", err, wantErr)
 	}
 }

